### PR TITLE
Practicie with a Mock Server and Changing Data on the DOM

### PR DESCRIPTION
**Pull request by Tomm1128**
The lab wanted us to practice getting back a successful server response and then manipulating the DOM. We also had to handle if the server gave us an error.

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -3,9 +3,38 @@ const EMPTY_HEART = '♡'
...
+    .catch(handleError)
+  } else {
+    mimicServerCall()
+    .then(() => {
+      errorBlock.className = "hidden"
+      event.target.className = "like-glyph"
+      event.target.innerText = EMPTY_HEART
+    })
+    .catch(handleError)
+  }
```
Comment (stephenmckeon) [inline main.js]: Can you think of a way to refactor this to have only one call to `mimicServerCall`?

**Review by stephenmckeon — APPROVED** [review]

File: main.js (hunk 1 above)
Comment (Tomm1128) [reply main.js]: Would I make the server call, then do my if statement and run the then methods in there? So roughly something like 
:
```js
mimicServerCall() 
 if(event.target.innerText === EMPTY_HEART){
    .then(() => {
      errorBlock.className = "hidden"
      event.target.className = "activated-heart" 
      event.target.innerText = FULL_HEART
    })
    .catch(handleError)
} else {
    .then(() => {
      errorBlock.className = "hidden"
      event.target.className = "like-glyph"
      event.target.innerText = EMPTY_HEART
    })
    .catch(handleError)
  }
}
```

Comment (stephenmckeon) [reply main.js]: Not quite. You can reduce the `.then`s and you can try moving the `if` statement around. If you have some extra time, come back and try a few things out.

**Comment by Tomm1128** [comment]
Hey @stephenmckeon , I refactored the code to reduce the .then (s) and only make one server call. Let me know what you think

**Review by stephenmckeon — APPROVED** [review]
That's exactly what I was looking for! Nice job.

**Pull request closed without merging**